corehq/apps/analytics/tests/test_partner_analytics_utils.py

```diff
@@ -0,0 +1,385 @@
+import calendar
+import datetime
+
+from unittest import mock
+
+from django.test import TestCase
+
+from corehq.apps.accounting.models import DomainUserHistory
+from corehq.apps.analytics.models import (
+    PartnerAnalyticsContact,
+    PartnerAnalyticsReport,
+    PartnerAnalyticsDataPoint,
+)
+from corehq.apps.analytics.utils.partner_analytics import (
+    get_number_of_mobile_workers,
+    get_number_of_web_users,
+    get_number_of_submissions,
+    generate_monthly_mobile_worker_statistics,
+    track_partner_access,
+    generate_monthly_web_user_statistics,
+    generate_monthly_submissions_statistics,
+    get_csv_details_for_partner,
+    send_partner_emails,
+    NUMBER_OF_MOBILE_WORKERS,
+    NUMBER_OF_WEB_USERS,
+    NUMBER_OF_SUBMISSIONS,
+    ACCESS_ODATA,
+)
+from corehq.apps.domain.models import Domain
+from corehq.apps.users.models import WebUser, Invitation
+from corehq.elastic import get_es_new, send_to_elasticsearch
+from corehq.form_processor.tests.utils import (
+    FormProcessorTestUtils,
+    create_form_for_test,
+)
+from corehq.pillows.mappings import USER_INDEX_INFO, XFORM_INDEX_INFO
+from corehq.pillows.user import transform_user_for_elasticsearch
+from corehq.pillows.xform import transform_xform_for_elasticsearch
+from corehq.util.elastic import ensure_index_deleted
+from dimagi.utils.dates import get_start_and_end_dates_of_month
+from pillowtop.es_utils import initialize_index_and_mapping
+
+
+def _get_fake_number_of_mobile_workers(domain, _year, _month):
+    return {
+        'test-001': 10,
+        'test-002': 20,
+        'test-003': 30,
+    }[domain]
+
+
+def _get_fake_number_of_web_users(domain, _year, _month):
+    return {
+        'test-001': 11,
+        'test-002': 21,
+        'test-003': 31,
+    }[domain]
+
+
+def _get_fake_number_of_submissions(domain, _year, _month):
+    return {
+        'test-001': 11,
+        'test-002': 22,
+        'test-003': 33,
+    }[domain]
+
+
+class TestPartnerAnalyticsDataUtils(TestCase):
```

Comment:
Sorry, in my original review I didn't realize this was the class where you needed to do all the Elastic stuff. I thought it was the other one (which you told me already).  This should have the `es_test` decorator now.